internal/builder/builder.go

```diff
@@ -417,6 +417,37 @@ func addBuildpacks(logger logging.Logger, tmpDir string, image imgutil.Image, ad
 			if existingBPInfo.LayerDiffID == diffID.String() {
 				logger.Debugf("Buildpack %s already exists on builder with same contents, skipping...", style.Symbol(bpInfo.FullName()))
 				continue
+			} else {
+
+				bpWhiteoutsTmpDir := filepath.Join(tmpDir, strconv.Itoa(i)+"_whiteouts")
+				if err := os.MkdirAll(bpWhiteoutsTmpDir, os.ModePerm); err != nil {
+					return errors.Wrap(err, "creating buildpack whiteouts temp dir")
+				}
+
+				deletedMaps := map[string][]byte{
+					filepath.Join(
+						cnbDir,
+						"buildpacks",
+						strings.ReplaceAll(bpInfo.ID, "/", "_"),
+						".wh."+bpInfo.Version,
+					): {},
+				}
+				whiteoutsTarFile := filepath.Join(bpWhiteoutsTmpDir, "whiteouts.tar")
+
+				if err := createTarball(whiteoutsTarFile, deletedMaps); err != nil {
+					return errors.Wrapf(err,
+						"creating whiteout layers' tarfile for buildpack %s",
+						style.Symbol(bp.Descriptor().Info.FullName()),
+					)
+				}
+
+				if err := image.AddLayer(whiteoutsTarFile); err != nil {
+					return errors.Wrapf(err,
+						"adding whiteout layer tar for buildpack %s",
+						style.Symbol(bpInfo.FullName()),
+					)
+				}
+
```

Comment:
nit: Can you remove this line?

Reply:
I changed that line, but I doubt I got it right :)